### PR TITLE
Cleanup: move toFnWithContext() to CombineFnUtil

### PR DESCRIPTION
**Pull request by peihe**

### CODE REVIEW

**Comment by peihe** [comment]
R: @bjchambers

**Comment by bjchambers** [comment]
Is there any way to unit test the adaptor?

Also, I see this failing in the travis builds:
https://travis-ci.org/apache/incubator-beam/jobs/116767036

**Comment by kennknowles** [comment]
Need to resolve merge conflicts.

**Comment by peihe** [comment]
merged with master

**Comment by bjchambers** [comment]
It seems like it should be possible to add some tests for the adaptor.

File: sdks/java/core/src/test/java/com/google/cloud/dataflow/sdk/util/CombineFnUtilTest.java [hunk 1]
```diff
@@ -59,4 +62,15 @@ public void testNonSerializable() throws Exception {
...
+
+  @Test
+  public void testToFnWithContext() throws Exception {
+    CombineFnWithContext<Integer, int[], Integer> fnWithContext =
+        CombineFnUtil.toFnWithContext(new Sum.SumIntegerFn());
+    assertTrue(fnWithContext == CombineFnUtil.toFnWithContext(fnWithContext));
+
+    KeyedCombineFnWithContext<Object, Integer, int[], Integer> keyedFnWithContext =
+        CombineFnUtil.toFnWithContext(new Sum.SumIntegerFn().asKeyedFn());
+    assertTrue(keyedFnWithContext == CombineFnUtil.toFnWithContext(keyedFnWithContext));
```
Comment (bjchambers) [inline sdks/java/core/src/test/java/com/google/cloud/dataflow/sdk/util/CombineFnUtilTest.java]: This is only testing that the conversion is idempotent. We should also test that the generated wrapper actually works (eg., use the wrapper produced from Sum and make sure it works by exercising all the methods).

Comment (peihe) [inline sdks/java/core/src/test/java/com/google/cloud/dataflow/sdk/util/CombineFnUtilTest.java]: done

**Comment by bjchambers** [comment]
LGTM

**Pull request closed without merging**